woeip/apps/air_quality/tests/factories.py

```diff
@@ -9,10 +9,20 @@
 from woeip.apps.core.tests.factories import UserFactory
 
 
-class FuzzyLineString(factory.fuzzy.BaseFuzzyAttribute):
+def random_latlon():
```

Comment:
There is no reason to reinvent this. Use `faker`: https://faker.readthedocs.io/en/master/providers/faker.providers.geo.html.

Reply:
The faker output is a tuple of ``decimal.Decimal``, which isn't compatible with the ``django.contrib.gis.db.models.PointField``. It really really wants the value to be a ``django.contrib.gis.geos.Point``.

Reply:
Specifically...

```
>>> data = models.Data(session=session, value=2.3, time=datetime.datetime.now(), latlon=fake.latlng())
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/django/db/models/base.py", line 470, in __init__
    _setattr(self, field.attname, val)
  File "/usr/local/lib/python3.6/site-packages/django/contrib/gis/db/models/proxy.py", line 76, in __set__
    instance.__class__.__name__, gtype, type(value)))
TypeError: Cannot set Data SpatialProxy (POINT) with value of type: <class 'tuple'>
```

Reply:
I see, you were referring to something else. Addressed below!